packages/clerk-js/src/ui/elements/PhoneInput/PhoneInput.tsx

```diff
@@ -37,9 +42,48 @@ export const PhoneInput = (props: PhoneInputProps) => {
   const selectedCountryOption = React.useMemo(() => {
     return countryOptions.find(o => o.country.iso === selectedIso) || countryOptions[0];
   }, [selectedIso]);
+  const dynamicPadding = selectedCountryOption.country.code.length * 5; // this to calculate the padding of the input field depending the length of country code
 
   React.useEffect(callOnChangeProp, [cleanPhoneNumber]);
 
+  const handlePaste = (e: ClipboardEvent<HTMLInputElement>) => {
+    const inputValue = e.clipboardData.getData('text');
+    const testPattern = /\+/g;
+
+    if (testPattern.test(inputValue)) {
+      setHasBeenPasted(true);
+
+      let result = '';
+      let selectedCountry = null;
+      for (let i = 5; i > 1; i--) {
+        const code = inputValue.slice(1, i);
+
+        result = inputValue.slice(i, inputValue.length);
+        selectedCountry = countryOptions.find(o => o.country.code === code);
+
+        if (selectedCountry) {
+          break;
+        }
+      }
+
+      if (selectedCountry) {
+        setSelectedIso(selectedCountry?.country.iso);
+        setPhoneNumber(result, selectedCountry?.country.iso);
+        return;
+      }
+    }
+    setSelectedIso(selectedIso);
+    setPhoneNumber(inputValue);
```

Comment:
🚫 I'm pretty sure these two lines need to be removed since normal pastes(without the country code) are being duplicated. If removed, the `setPhoneNumber` will run in the `onChange` handler since the `hasBeenPasted` flag would be false. Not sure why the `setSelectedIso` was added here. Am I missing something?

Reply:
@desiprisg The `setPhoneNumber` is needed here if the phone number(with county code) does not correspond to a valid county code. So you just paste it as is and let the API inform the user that this is not valid.

The `setSelectedIso` seems to be redundant, you are right🙃

Reply:
Still, when pasting something that does not satisfy the regular expression, we get a double paste right now!

Can you give me an example input to paste to see the above behaviour?

Reply:
@desiprisg why would we get a double-paste?

For example this is not valid `+699090909090`

Reply:
We get a double-paste when pasting strings like `123`, strings that fail the test pattern! Does that not happen on your machine?

Reply:
@desiprisg oh I see what you mean. I forgot to move the `setPhoneNumber` within the test pattern check. Now it does not produce a double check

Reply:
@desiprisg 

https://user-images.githubusercontent.com/23478420/198993933-662495f4-13d0-486c-a0c6-778678c25899.mov

Reply:
@raptisj @desiprisg hey folks, is this one resolved?

Reply:
Yes, seems fine in the vid. Sorry it took so long, I missed the notification. We can resolve this one. I'll make another pass anyway!